query-engine/js-connectors/src/proxy.rs

```diff
@@ -289,16 +294,143 @@ fn js_planetscale_value_to_quaint(json_value: serde_json::Value, column_type: Co
+        },
+        ColumnType::Time => match json_value {
+            serde_json::Value::String(s) => {
+                let time = NaiveTime::parse_from_str(&s, "%H:%M:%S").expect("Expected a time string");
+                QuaintValue::time(time)
+            }
+            serde_json::Value::Null => QuaintValue::Time(None),
+            mismatch => panic!("Expected a string, found {:?}", mismatch),
+        },
+        ColumnType::DateTime => match json_value {
+            serde_json::Value::String(s) => {
+                let datetime = chrono::NaiveDateTime::parse_from_str(&s, "%Y-%m-%d %H:%M:%S")
+                    .unwrap_or_else(|_| panic!("Expected a datetime string, found {:?}", &s));
+                let datetime: DateTime<Utc> = DateTime::from_utc(datetime, Utc);
+                QuaintValue::datetime(datetime)
+            }
+            serde_json::Value::Null => QuaintValue::DateTime(None),
+            mismatch => panic!("Expected a string, found {:?}", mismatch),
+        },
+        ColumnType::Json => match json_value {
+            serde_json::Value::Null => QuaintValue::Json(None),
+            json => QuaintValue::json(json),
+        },
+        ColumnType::Enum => match json_value {
+            serde_json::Value::String(s) => QuaintValue::enum_variant(s),
+            serde_json::Value::Null => QuaintValue::Enum(None),
+            mismatch => panic!("Expected a string, found {:?}", mismatch),
+        },
+        ColumnType::Bytes => match json_value {
+            serde_json::Value::String(s) => QuaintValue::Bytes(Some(s.into_bytes().into())),
+            serde_json::Value::Null => QuaintValue::Bytes(None),
+            mismatch => panic!("Expected a string, found {:?}", mismatch),
+        },
+        unimplemented => {
+            todo!("support column type: Column: {:?}", unimplemented)
+        }
+    }
+}
+
+pub struct FlavoredJSResultSet(pub (JsConnectorFlavor, JSResultSet));
+
+impl From<FlavoredJSResultSet> for QuaintResultSet {
+    fn from(pair: FlavoredJSResultSet) -> Self {
+        let (flavor, mut js_result_set) = pair.0;
         // TODO: extract, todo: error rather than panic?
         let to_quaint_row = move |row: &mut Vec<serde_json::Value>| -> Vec<quaint::Value<'static>> {
             let mut res = Vec::with_capacity(row.len());
 
             for i in 0..row.len() {
                 let column_type = js_result_set.column_types[i];
                 let json_value = row.remove(0);
-                let quaint_value = js_planetscale_value_to_quaint(json_value, column_type);
+
+                // Note: here, we could consider using conditional compile-time variables to avoid the match.
+                let quaint_value = match flavor {
+                    JsConnectorFlavor::MySQL => js_planetscale_value_to_quaint(json_value, column_type),
+                    JsConnectorFlavor::Postgres => js_neon_value_to_quaint(json_value, column_type),
```

Comment:
This might be problematic in the future, because we are assuming there's only one implementation per driver. Also I don't feel too happy that we are doing driver-specific (as in planetscale and neon) logic in the rust land. 

We openeed an issue https://github.com/prisma/team-orm/issues/257 to offload this parsing logic to javascript. So right now, we get data from the driver as binary tuples  containing a type, and a value. Those are to represent a Rust enum, which is a tagged enum, with non-unit variants. We do it, because there is no 1:1 translation of that abstraction into javascript. But the only purpose of those tuples is to be able to reconstruct values of the Quaint::Value type. As such, we can perform all the conversions necessary in the driver, thus conversion logic that is specific to the driver within the own driver and not scattered accross the engine as well. 

Once done that, the current implementation of js_*_value_to_quaint, will be only one in rust, and only transform the shape coming from the different driver implementations (type, value) -> Quiant::Value